third_party/libcxx/libcxx.mk

```diff
@@ -212,6 +214,16 @@ $(THIRD_PARTY_LIBCXX_A_OBJS): private				\
 			-ffunction-sections			\
 			-fdata-sections
 
+# Super gross! But I found no way to do -fno-error=permissive
+# this steps from the fact that demangle is a C++ file extension
+# but the source I copied was written in C.
+# The fix would be to go through all the casts and make them correct.
+o/$(MODE)/third_party/libcxx/demangle.o: private				\
```

Comment:
Two options here:
1. is there a way to compile this as a C file while still including the other libcxx files?
I guess I could include the libc equivalent and just change this to a .c file?

2. edit the source to add casts to not emit the permissive error anymore.

Reply:
With something like this, the dependency footprint is so small that the best place for it is probably libc/stdio/. I just plopped the file in there, made sure the license was at the top, and then added a single line to stdio.h.

![image](https://github.com/jart/cosmopolitan/assets/49262/fadf537e-ecf1-45c4-ad84-092a0526ee4c)

Those `COSMOPOLITAN_C_END_` etc. defines are what automate the `extern "C"` thing, so that C++ code can include C headers. I'm going to cherry pick this change onto the master branch manually in a moment.